### ISSUE TITLE
[BUG] Cannot create EKS cluster Field [nodeRole] cannot be nil for nodegroup [ng] in non-nil cluster [c-qm5db]

### ISSUE THREAD

**Issue by sanjay920** [rancher/rancher#41123]
**Rancher Server Setup**
- Rancher version: 2.7.1
- Installation option (Docker install/Helm Chart): Helm chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): EKS
- Proxy/Cert Details: LetsEncrypt

**Information about the Cluster**
- Kubernetes version: 1.23
- Cluster Type (Local/Downstream): Downstream EKS cluster

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**

Rancher fails to create an EKS cluster. Cluster created through the Rancher UI. AWS resources are not created (I checked in aws) and the following error message is displayed for the cluster
```
This cluster is currently Provisioning; areas that interact directly with it will not be available until the API is ready.

Field [nodeRole] cannot be nil for nodegroup [sanjay-test-ng] in non-nil cluster [c-qm5db]
```

**To Reproduce**

Create an EKS cluster

Account Access:
* Region: any region (I have tried us-east-1 and us-west-2)

Cluster Options
* Kubernetes version 1.23
* Service Role: Rancher created role
* Everything else default

Networking
* API server endpoint access: Public access only
* Rancher created VPC and subnet

Security Group:
* Auto created security group

Managed Node Groups:
Only 1 node group with the following:
* No launch template given
* Instance type t3.medium
* No AMI ID provided
* Node volume size 20gb
* No ssh key provided
* Desired ASG Size 2
* Maximum ASG Size 2
* Minimum ASG Size 2

**Result**
EKS operator logs
```
time="2023-04-10T23:57:06Z" level=error msg="Error removing metadata from failure message: message body not formatted as expected"
time="2023-04-10T23:57:06Z" level=error msg="error syncing 'cattle-global-data/c-qm5db': handler eks-controller: field [nodeRole] cannot be nil for nodegroup [sanjay-test-ng] in non-nil cluster [c-qm5db], requeuing"
time="2023-04-10T23:59:07Z" level=error msg="Error removing metadata from failure message: message body not formatted as expected"
time="2023-04-10T23:59:07Z" level=error msg="error syncing 'cattle-global-data/c-qm5db': handler eks-controller: field [nodeRole] cannot be nil for nodegroup [sanjay-test-ng] in non-nil cluster [c-qm5db], requeuing"
time="2023-04-11T00:01:07Z" level=error msg="Error removing metadata from failure message: message body not formatted as expected"
time="2023-04-11T00:01:07Z" level=error msg="error syncing 'cattle-global-data/c-qm5db': handler eks-controller: field [nodeRole] cannot be nil for nodegroup [sanjay-test-ng] in non-nil cluster [c-qm5db], requeuing"
time="2023-04-11T00:03:07Z" level=error msg="Error removing metadata from failure message: message body not formatted as expected"
time="2023-04-11T00:03:07Z" level=error msg="error syncing 'cattle-global-data/c-qm5db': handler eks-controller: field [nodeRole] cannot be nil for nodegroup [sanjay-test-ng] in non-nil cluster [c-qm5db], requeuing"
time="2023-04-11T00:05:08Z" level=error msg="Error removing metadata from failure message: message body not formatted as expected"
time="2023-04-11T00:05:08Z" level=error msg="error syncing 'cattle-global-data/c-qm5db': handler eks-controller: field [nodeRole] cannot be nil for nodegroup [sanjay-test-ng] in non-nil cluster [c-qm5db], requeuing"
```

Rancher logs:
```
2023/04/10 23:56:49 [ERROR] failed to start cluster controllers c-m-zfk5chqb: context canceled
--
Mon, Apr 10 2023 4:58:42 pm | 2023/04/10 23:58:42 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 4:58:43 pm | 2023/04/10 23:58:43 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-mvs8dx5b from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 4:58:44 pm | 2023/04/10 23:58:44 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.765941 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767288 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767476 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767476 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767398 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767515 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767670 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767807 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767929 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.768078 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.768165 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.768337 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767302 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.767754 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:51 pm | W0410 23:58:51.947813 33 transport.go:288] Unable to cancel request for *client.addQuery
Mon, Apr 10 2023 4:58:58 pm | 2023/04/10 23:58:58 [INFO] Stopping cluster agent for c-m-zfk5chqb
Mon, Apr 10 2023 4:58:58 pm | 2023/04/10 23:58:58 [ERROR] failed to start cluster controllers c-m-zfk5chqb: context canceled
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:00 pm | 2023/04/11 00:00:00 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:06 pm | 2023/04/11 00:00:06 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-mvs8dx5b from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:06 pm | 2023/04/11 00:00:06 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-mvs8dx5b from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:12 pm | W0411 00:00:12.622917 33 warnings.go:80] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
Mon, Apr 10 2023 5:00:42 pm | 2023/04/11 00:00:42 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:43 pm | 2023/04/11 00:00:43 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-mvs8dx5b from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:44 pm | 2023/04/11 00:00:44 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:00:57 pm | 2023/04/11 00:00:57 [INFO] Stopping cluster agent for c-m-zfk5chqb
Mon, Apr 10 2023 5:00:57 pm | 2023/04/11 00:00:57 [ERROR] failed to start cluster controllers c-m-zfk5chqb: context canceled
Mon, Apr 10 2023 5:02:42 pm | 2023/04/11 00:02:42 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:02:43 pm | 2023/04/11 00:02:43 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-mvs8dx5b from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:02:44 pm | 2023/04/11 00:02:44 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:02:53 pm | 2023/04/11 00:02:53 [INFO] Stopping cluster agent for c-m-zfk5chqb
Mon, Apr 10 2023 5:02:53 pm | 2023/04/11 00:02:53 [ERROR] failed to start cluster controllers c-m-zfk5chqb: context canceled
Mon, Apr 10 2023 5:04:36 pm | 2023/04/11 00:04:36 [INFO] Stopping cluster agent for c-m-zfk5chqb
Mon, Apr 10 2023 5:04:36 pm | 2023/04/11 00:04:36 [ERROR] failed to start cluster controllers c-m-zfk5chqb: context canceled
Mon, Apr 10 2023 5:04:42 pm | 2023/04/11 00:04:42 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:04:43 pm | 2023/04/11 00:04:43 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-mvs8dx5b from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:04:44 pm | 2023/04/11 00:04:44 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:06:42 pm | 2023/04/11 00:06:42 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-nqq4cr85 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:06:43 pm | 2023/04/11 00:06:43 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-mvs8dx5b from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:06:44 pm | 2023/04/11 00:06:44 [ERROR] error syncing 'kube-system/k3s-etcd-snapshots': handler snapshotbackpopulate: error while retrieving cluster c-m-ccfgtqf8 from cache via index: Index with name by-cluster does not exist, requeuing
Mon, Apr 10 2023 5:07:14 pm | 2023/04/11 00:07:14 [INFO] Stopping cluster agent for c-m-zfk5chqb
Mon, Apr 10 2023 5:07:14 pm | 2023/04/11 00:07:14 [ERROR] failed to start cluster controllers c-m-zfk5chqb: context canceled


```

**Expected Result**

Create EKS cluster via Rancher UI successfully

**Screenshots**

**Additional context**

It is worth noting that this issue only started appearing on April 7, 2023. Prior to then, I didn't experience this issue when creating EKS clusters. The rancher instance being used has been up for the last 5 months.

**Comment by pgonin** [comment]
So the root cause appears to be that a new version of the UI in Rancher 2.7.2 sets nodeRole as non-nil which is required with eks-operator v1.2.0 to works correctly (on Rancher 2.7.2)
for details see
https://github.com/rancher/dashboard/issues/4568#issuecomment-1297270190
https://github.com/rancher/eks-operator/pull/73
The new chart for eks-operator (102.0.0+up1.2.0) that was released on April 7th (as part of preparation of 2.7.2 release) set the rancher-version annotation to
`  [catalog.cattle.io/rancher-version](http://catalog.cattle.io/rancher-version): '>= 2.7.0-0 < 2.8.0-0'`
with the consequence that eks-operator v1.2.0 gets deployed on Rancher 2.7.0 and Rancher 2.7.1 including existing deployments and breaks EKS provisionning.

We're working on a workaround and a fix

**Comment by zackbradys** [comment]
I'm seeing the same issues as well. I have tried multiple fresh clusters and the issue appears each time.

**Comment by furkatgofurov7** [comment]
Hey,

Please upgrade to the latest version of Rancher (2.7.2) and eks-operator (v1.2.0) to make it work, we have tested it and it works for the above versions.

**Comment by pgonin** [comment]
if you can't or do not want to upgrade to 2.7.2 yet, here is a temporary workaround:
apply on your Rancher local cluster:
```kubectl set image -n cattle-system  deployment/eks-config-operator eks-operator=rancher/eks-operator:v1.1.5```

**Comment by zackbradys** [comment]
We're seeing this issue as well. 

Upgrading to MCM 2.7.2 or reverting back to the EKS Operator 1.1.5 fixed it. Appreciate the quick response @pgonin

**Comment by pgonin** [comment]
Also, airgapped deployments are not impacted as the image mirrored from the 2.7.1 list of assets is `docker.io/rancher/eks-operator:v1.1.5 sha256:28d66230bdf850e9ec3a996553e3a667e3f79cce0a1472ca0ae71f2fe3c2e88b`

As soon as the registry used for airgap deployments is updated with 2.7.2 assets, the issue may arise

**Comment by kkaempf** [comment]
@sowmyav27 - fix is ready to test

Rancher PR: https://github.com/rancher/rancher/pull/41261
Charts PR: https://github.com/rancher/charts/pull/2567

**Comment by maradwan** [comment]
I am still facing the issue when I used terraform, it works fine with Rancher UI.

**Comment by cpinjani** [comment]
Waiting for PR https://github.com/rancher/charts/pull/2693 to get merged to validate this

**Comment by cpinjani** [comment]
**Ticket https://github.com/rancher/rancher/issues/41123 - Test Results**

EKS Operator version:
> rancher/eks-operator:v1.2.0

Test case:
> Provision downstream EKS Cluster with Rancher UI, steps as given [here](https://github.com/rancher/rancher/issues/41123#issue-1661575157)

Rancher version | Result |
-- | -- |
2.7.0|:negative_squared_cross_mark: Field [nodeRole] cannot be nil for nodegroup [test] in non-nil cluster [c-q8h4c]
2.7.1|:negative_squared_cross_mark: Field [nodeRole] cannot be nil for nodegroup [test] in non-nil cluster [c-n9gld]
2.7.2|:white_check_mark: Pass 
2.7.3|:white_check_mark: Pass

@alexander-demicev Can you please check if PR https://github.com/rancher/eks-operator/pull/138 needs backporting or a new patch to fix this ?

**Comment by alexander-demicev** [comment]
@cpinjani I think the bug fix has to be backported

**Comment by davidstauffer** [comment]
Hi, the fix is available from 2.7.2 onwards. We are currently not planning to backport the fix to prior versions. CC @kkaempf

**Comment by kkaempf** [comment]
Closing as done based on the verification [done last week](https://github.com/rancher/rancher/issues/41123#issuecomment-1577036222)
